pkg/generated/restapi/configure_fulcio_server.go

```diff
@@ -95,10 +98,29 @@ func configureAPI(api *operations.FulcioServerAPI) http.Handler {
 		if idToken == nil {
 			return nil, goaerrors.New(http.StatusUnauthorized, strings.Join(errs, ","))
 		}
+		if _, ok, err := oauthflow.EmailFromIDToken(idToken); !ok || err != nil {
+			if err != nil {
+				return nil, goaerrors.New(http.StatusUnauthorized, err.Error())
+			}
+			return nil, goaerrors.New(http.StatusUnauthorized, "email has not been verified with token issuer")
+		}
 
 		return idToken, nil
 	}
-	api.SigningCertHandler = operations.SigningCertHandlerFunc(pkgapi.SigningCertHandler)
+
+	// Select which CA / KMS system to use
+	// Currently supported:
+	// googleca: Google Certficate Authority Service
+	// fulcio: Generic PKCS11 / HSM backed service
+	CAType := viper.GetString("ca")
+	switch CAType {
+	case "googleca":
+		api.SigningCertHandler = operations.SigningCertHandlerFunc(pkgapi.GoogleCASigningCertHandler)
+	case "fulcioca":
+		api.SigningCertHandler = operations.SigningCertHandlerFunc(pkgapi.FulcioCASigningCertHandler)
+	default:
```

Comment:
if it is neither, should we bail since we don't have the information to actually interact with the googleca?

Reply:
good call!